base/v0_6_exp/validate.go

```diff
@@ -41,9 +44,39 @@ func (rs Resource) Validate(c path.ContextPath) (r report.Report) {
 	if sources > 1 {
 		r.AddOnError(c.Append(field), common.ErrTooManyResourceSources)
 	}
+	if field == "local" || field == "inline" {
+		// check if it's an ignition config
+		if len(config) > 0 && config[0] == '{' {
+			_, report, err := exp.Parse([]byte(config))
+			if len(report.Entries) > 0 {
+				butaneReport = MapIgnitionReportToButane(report)
+				r.Merge(butaneReport)
+			}
+			if err != nil {
+				if err == common.ErrNoFilesDir {
+					r.AddOnError(c.Append(field), common.ErrNoFilesDir)
+				}
+			}
+		}
+	}
+
 	return
 }
 
+func MapIgnitionReportToButane(ignitionReport report.Report) report.Report {
+	var butaneRep report.Report
+	for _, entry := range ignitionReport.Entries {
+		butaneEntry := report.Entry{
```

Comment:
Im not sure this is everything we need.
Looping through the entries and expanding them into a new report object with those entries is good.
The issue comes with the the entry.Context path the paths are different between the ignition package and the butane package. this is mostly because we are converting from json to yaml butane is human readable so their names change.

internal to butane it looks like we already have a similar issue, we have to translate to json from yaml the direction is the inverse of what we need now though.  

It has a good test to explain this here => https://github.com/coreos/butane/blob/0244d31c627890b9af64b82e2a2224ac39bbb4d4/base/v0_6_exp/translate_test.go#L573C12-L573C12